pkg/util/parser.go

```diff
@@ -10,34 +10,35 @@ import (
 	"strings"
 )
 
-func EntryParser(entry, delimiter string) (string, error) {
+func getIndex(input string) int {
+	idx1 := strings.Index(input, ",")
+	idx2 := strings.Index(input, "/")
 
-	if len(entry) < 1 {
-		return "", fmt.Errorf("invalid length")
+	if idx1 == -1 {
+		return idx2
 	}
+	return idx1
+}
+
+func ReadField(input string) (string, int, error) {
 
-	size := len(entry) - 1
-	if entry[size:] != delimiter {
-		return "", fmt.Errorf("contains invalid delimiter")
+	idx := getIndex(input)
+	if idx == -1 {
+		return "", 0, fmt.Errorf("doesn't have valid delimiter")
 	}
 
-	return entry[:size], nil
+	return input[:idx], idx + 1, nil
 }
 
-func EntryParserToInt(entry, delimiter string) (int64, error) {
+func ReadFieldAsInt(input string) (int64, int, error) {
 
-	if len(entry) < 1 {
-		return 0, fmt.Errorf("invalid length")
+	idx := getIndex(input)
+	if idx == -1 {
+		return 0, 0, fmt.Errorf("doesn't have valid delimiter")
 	}
 
-	size := len(entry) - 1
-	if entry[size:] != delimiter {
-		return 0, fmt.Errorf("contains invalid delimiter")
-	}
-
-	value, _ := strconv.ParseInt(entry[:size], 10, 64)
-
-	return value, nil
+	value, _ := strconv.ParseInt(input[:idx], 10, 64)
+	return value, idx + 1, nil
 }
 
 func GetSize(line string) int64 {
```

Comment:
Does this still produce correct sizes with all allowed characters? We [ran into `'` being counted incorrectly](https://github.com/moov-io/ach/issues/1156).

Reply:
ReadField, ReadFieldAsInt will be get string/int value from input line
delimiters are '/' or ','
it seems that we don't need to check one/two byte characters

Reply:
Can we add some tests to make sure the code doesn't panic or return unexpected results?

Reply:
Just noticed this – we absolutely shouldn't be ignoring errors from calling `ParseInt` on user supplied input. We should return the error (or a wrapped one) instead. It won't panic on malformed input, but possibly worse, it will fail silently and return `0` to the caller.